### PR TITLE
Misc clerk fixes

### LINKED ISSUE #34: Line items are not ordered by most recent
When products are added to the sale, the line items should be added to the bottom of the Current Sale list but instead, it seems to be ordered alphabetically.

### LINKED ISSUE #56: ProductDisplayCard warning
Getting this warning on checkout screen. Should be fixable by adding boolean check since not all products might have details.
- ```Warning: Failed prop type: The prop `product.detail` is marked as required in `ProductDisplayCard`, but its value is `undefined`.```

### PR DESCRIPTION
**Pull request by thumn**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR
- Changed `ProductDisplayCard` propTypes to be un-required (since it's possible for some products to not have a name, image, and/or detail line) and provided default props. This removes a warning on `CheckoutScreen`.
- Line items are no longer alphabetized -- now ordered by most recent in the cart.

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

## Relevant Links
Closes #56 
Closes #34 

### Online sources

[//]: # 'Optional - copy links to any tutorial or documentation that was useful to you when working on this PR'

### Related PRs

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

## How to review
You shouldn't see the propTypes warning on the checkout screen anymore. Also, the "Current Sale" section should function as normal, but items most recently added should appear at the bottom of the list. Try adding and removing items.

[//]: # 'The order in which to review files and what to expect when testing locally'

## Next steps

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"

## Tests Performed, Edge Cases

[//]: # 'Hopefully we will add a testing suite/CI soon, but until then note down the steps you took to test locally'

### Screenshots
IMG_2120

[//]: # "Add screenshots of expected behavior - GIFs if you're feeling fancy!"

CC: @anniero98 @wangannie

[//]: # 'This tags in both Annies as a default. Feel free to change, or add on anyone who you should be in on the conversation.'

### CODE REVIEW

File: screens/CheckoutScreen.js [hunk 1]
```diff
@@ -72,6 +66,17 @@ export default class CheckoutScreen extends React.Component {
       2) No rewards have been applied, or
       3) The new balance is non-negative
       No special handling */
+    // Add a new product to lineItems.
```
Comment (anniero98) [inline screens/CheckoutScreen.js:69]: very clean ✨
